File: automator/src/main/java/com/stardust/automator/filter/IdFilter.java
package com.stardust.automator.filter;

import com.stardust.automator.UiObject;

import java.util.List;

/**
 * Created by Stardust on 2017/3/9.
 */

public class IdFilter {

    public interface FullIdGetter {
        String getFullId(String id);
    }

    private static final KeyGetter ID_GETTER = new KeyGetter() {

        @Override
        public String getKey(UiObject nodeInfo) {
            return nodeInfo.getViewIdResourceName();
        }

        @Override
        public String toString() {
            return "id";
        }
    };

    public static StringEqualsFilter equals(String id) {
        return new StringEqualsFilter(id, ID_GETTER);
    }
    public static StringStartsWithFilter startsWith(String prefix) {
        return new StringStartsWithFilter(prefix, ID_GETTER);
    }

    public static ListFilter endsWith(String suffix) {
        return new StringEndsWithFilter(suffix, ID_GETTER);
    }

    public static StringContainsFilter contains(String contains) {
        return new StringContainsFilter(contains, ID_GETTER);
    }

    public static StringMatchesFilter matches(String regex) {
        return new StringMatchesFilter(regex, ID_GETTER);
    }

}

File: automator/src/main/java/com/stardust/automator/filter/IdFilter.kt
package com.stardust.automator.filter

import com.stardust.automator.UiObject

/**
 * Created by Stardust on 2017/3/9.
 */

object IdFilter {

    private val ID_GETTER = object : KeyGetter {

        override fun getKey(nodeInfo: UiObject): String? {
            return nodeInfo.viewIdResourceName
        }

        override fun toString(): String {
            return "id"
        }
    }

    fun equals(id: String): StringEqualsFilter {
        return StringEqualsFilter(id, ID_GETTER)
    }

    fun startsWith(prefix: String): StringStartsWithFilter {
        return StringStartsWithFilter(prefix, ID_GETTER)
    }

    fun endsWith(suffix: String): StringEndsWithFilter {
        return StringEndsWithFilter(suffix, ID_GETTER)
    }

    fun contains(contains: String): StringContainsFilter {
        return StringContainsFilter(contains, ID_GETTER)
    }

    fun matches(regex: String): StringMatchesFilter {
        return StringMatchesFilter(regex, ID_GETTER)
    }

}
